barretenberg/cpp/src/barretenberg/eccvm/transcript_builder.hpp

```diff
@@ -104,225 +130,397 @@ class ECCVMTranscriptBuilder {
+        row.opcode = Opcode{ .add = entry.add, .mul = entry.mul, .eq = entry.eq, .reset = entry.reset }.value();
+    }
+
+    static void process_mul(const VMOperation& entry, VMState& updated_state, const VMState& state)
+    {
+
+        const auto P = typename CycleGroup::element(entry.base_point);
+        const auto R = typename CycleGroup::element(state.msm_accumulator);
+        updated_state.msm_accumulator = R + P * entry.mul_scalar_full;
+    }
+
+    static void process_add(const VMOperation& entry, VMState& updated_state, const VMState& state)
+    {
+
+        if (state.is_accumulator_empty) {
+            updated_state.accumulator = entry.base_point;
+        } else {
+            updated_state.accumulator = Element(state.accumulator) + entry.base_point;
+        }
+        updated_state.is_accumulator_empty = updated_state.accumulator.is_point_at_infinity();
+    }
+
+    static void process_msm_transition(TranscriptRow& row, VMState& updated_state, const VMState& state)
+    {
+        if (state.is_accumulator_empty) {
+            updated_state.accumulator = updated_state.msm_accumulator - offset_generator();
+        } else {
+            const Element R = Element(state.accumulator);
+            updated_state.accumulator = R + updated_state.msm_accumulator - offset_generator();
+        }
+        updated_state.is_accumulator_empty = updated_state.accumulator.is_point_at_infinity();
+
+        Element msm_output = updated_state.msm_accumulator - offset_generator();
+        row.transcript_msm_infinity = msm_output.is_point_at_infinity();
+    }
+    /**
+     * @brief Batched conversion of points in accumulators from Jacobian coordinates \f$ (X, Y, Z) \f$ to affine
+     * coordinates \f$ (x = X/Z^2, y = Y/Z^3 ) \f$.
+     *
+     * @param accumulator_trace
+     * @param msm_accumulator_trace
+     * @param intermediate_accumulator_trace
+     */
+    static void normalize_accumulators(Accumulator& accumulator_trace,
+                                       Accumulator& msm_accumulator_trace,
+                                       std::vector<Element>& intermediate_accumulator_trace)
+    {
         Element::batch_normalize(&accumulator_trace[0], accumulator_trace.size());
         Element::batch_normalize(&msm_accumulator_trace[0], msm_accumulator_trace.size());
         Element::batch_normalize(&intermediate_accumulator_trace[0], intermediate_accumulator_trace.size());
+    }
 
+    static void add_affine_coordinates_to_transcript(std::vector<TranscriptRow>& transcript_state,
+                                                     const Accumulator& accumulator_trace,
+                                                     const Accumulator& msm_accumulator_trace,
+                                                     const Accumulator& intermediate_accumulator_trace)
+    {
         for (size_t i = 0; i < accumulator_trace.size(); ++i) {
+            TranscriptRow& row = transcript_state[i + 1];
```

Comment:
this is not necessary, but makes the operations with the current transcipt row way clearer